### PR TITLE
(SDI-1977) End to end large test

### PR DESCRIPTION
**Pull request by rashmigottipati**
Summary of changes:
- Add scripts for large test
- Add examples for large test
- Add Dockerfile.erb

How to verify it:
- make test-large

### CODE REVIEW

**Comment by nanliu** [comment]
This is dependent on https://github.com/intelsdi-x/snap/pull/1349.

File: .sync.yml [hunk 1]
```diff
@@ -7,6 +7,37 @@
...
+  env:
+    global:
+      - ORG_PATH=/home/travis/gopath/src/github.com/intelsdi-x
+      - SNAP_PLUGIN_SOURCE=/home/travis/gopath/src/github.com/${TRAVIS_REPO_SLUG}
+    matrix:
+      - TEST_TYPE: small
+      - SNAP_VERSION: latest
+        SNAP_OS: alpine
+        TEST_TYPE: large
+      - SNAP_VERSION: latest_build
```
Comment (nanliu) [inline .sync.yml]: So not sure when we will release a new version of snap. For now only test on latest_build. When PR 1349 is merged, only latest_build will pass.

File: examples/tasks/psutil-file.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      config: 
+        /intel/mock: 
+          name: "root"
+          password: "secret"
+      process:
+        - plugin_name: "statistics"
+          publish:
+            - plugin_name: "file"
+              config:
+                 file: "/tmp/mock.log"
```
Comment (nanliu) [inline examples/tasks/psutil-file.yml]: Maybe use a more descriptive filename like psutil_stat.log

File: examples/tasks/psutil-file.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        /intel/psutil/load/load1: {}
+        /intel/psutil/load/load15: {}
+        /intel/psutil/load/load5: {}
+        /intel/psutil/vm/available: {}
+        /intel/psutil/vm/free: {}
+        /intel/psutil/vm/used: {} 
+      config: 
+        /intel/mock: 
+          name: "root"
+          password: "secret"
```
Comment (nanliu) [inline examples/tasks/psutil-file.yml]: Do we need a mock config since this is gathering from psutil?

File: scripts/spec/task_spec.rb [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          it {
+            id = subject.stdout.split("\n").find{|l|l=~/^ID:/}
+            task_id = $1 if id.match(/^ID: (.*)$/)
+            expect(task_id).to_not be_nil
+          }
+        end
+        
+        describe "snapctl task list" do
+          it {
+            sleep 15
```
Comment (nanliu) [inline scripts/spec/task_spec.rb]: We should add a comment on why we are sleeping here.

File: examples/tasks/statistics_file.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (nanliu) [inline examples/tasks/statistics_file.yml]: we should name this statistics-file.yml to be consistant with - vs _.

File: examples/tasks/statistics_file.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        /intel/psutil/load/load1: {}
+        /intel/psutil/load/load15: {}
+        /intel/psutil/load/load5: {}
+        /intel/psutil/vm/available: {}
+        /intel/psutil/vm/free: {}
+        /intel/psutil/vm/used: {} 
+      config: 
+        /intel/mock: 
+          name: "root"
+          password: "secret"
```
Comment (nanliu) [inline examples/tasks/statistics_file.yml]: same comment as above, is this necessary?

File: examples/tasks/statistics_file.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      process:
+        - plugin_name: "statistics"
+          config:
+            SlidingWindowLength: 5
+          publish:
+            - plugin_name: "file"
+              config:
+                 file: "/tmp/mock.log"
```
Comment (nanliu) [inline examples/tasks/statistics_file.yml]: same as log above

**Comment by candysmurf** [comment]
Steps to reproduce the error encountered:

1. cd snap-plugin-processor-statistics
2. make test-large

```
▶ make test-large
bash -c "./scripts/test.sh large"
2016-11-17 03:46:38 UTC [     info] spec folder : /Users/egu/.gvm/pkgsets/go1.7/snap-1.7/src/github.com/intelsdi-x/snap-plugin-processor-statistics/scripts
rbenv: bundle: command not found

The `bundle' command exists in these Ruby versions:
  2.3.1

make: *** [test-large] Error 127
```

3. I ran this command

```
rbenv local 2.3.1
```
test ran with partial errors:

```
Snap task statistics_file.yml
      Command "snapctl -u 'http://127.0.0.1:8181' task create -t /opt/snap/tasks/statistics_file.yml"
        should not be nil
        exit_status
          should eq 0
        stdout
          should contain /Task created/
      snapctl task list
        should contain /Running/ (FAILED - 1)
      Metrics in running tasks
        should include "/intel/psutil/vm/used"
      Stop task
        should eq 0 (FAILED - 2)
      Remove task
        should match /Task removed/

Failures:

  1) Docker Build from template: "/Users/egu/.gvm/pkgsets/go1.7/snap-1.7/src/github.com/intelsdi-x/snap-plugin-processor-statistics/examples/Dockerfile.erb" Serverspec on tag: "e67d05084339" Snap task statistics_file.yml snapctl task list should contain /Running/
     Failure/Error: expect(cmd.stdout).to contain /Running/
       expected "ID \t\t\t\t\t NAME \t\t\t\t\t\t STATE \t\t HIT \t MISS \t FAIL \t CREATED \t\t LAST FAILURE\n78b8ea3...ed \t 10 \t 0 \t 10 \t 3:51AM 11-17-2016 \t rpc error: code = 2 desc = Data is empty : Type <nil>\n" to contain /Running/

     # ./spec/task_spec.rb:55:in `block (6 levels) in <top (required)>'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

  2) Docker Build from template: "/Users/egu/.gvm/pkgsets/go1.7/snap-1.7/src/github.com/intelsdi-x/snap-plugin-processor-statistics/examples/Dockerfile.erb" Serverspec on tag: "e67d05084339" Snap task statistics_file.yml Stop task should eq 0
     Failure/Error: expect(c.exit_status).to eq 0

       expected: 0
            got: 1

       (compared using ==)

     # ./spec/task_spec.rb:80:in `block (6 levels) in <top (required)>'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # ./.bundle/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

Finished in 33.26 seconds (files took 28.65 seconds to load)
19 examples, 2 failures

Failed examples:

rspec './spec/task_spec.rb[1:1:5:2:1]' # Docker Build from template: "/Users/egu/.gvm/pkgsets/go1.7/snap-1.7/src/github.com/intelsdi-x/snap-plugin-processor-statistics/examples/Dockerfile.erb" Serverspec on tag: "e67d05084339" Snap task statistics_file.yml snapctl task list should contain /Running/
rspec './spec/task_spec.rb[1:1:5:4:1]' # Docker Build from template: "/Users/egu/.gvm/pkgsets/go1.7/snap-1.7/src/github.com/intelsdi-x/snap-plugin-processor-statistics/examples/Dockerfile.erb" Serverspec on tag: "e67d05084339" Snap task statistics_file.yml Stop task should eq 0

make: *** [test-large] Error 1

```

File: scripts/spec/task_spec.rb [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+          its(:stdout) { should contain /Task created/ }
+          it {
+            id = subject.stdout.split("\n").find{|l|l=~/^ID:/}
+            task_id = $1 if id.match(/^ID: (.*)$/)
+            expect(task_id).to_not be_nil
+          }
+        end
+       
+        describe "snapctl task list" do
+          #it { binding.pry }
```
Comment (nanliu) [inline scripts/spec/task_spec.rb]: Please remove this debug line.

**Comment by candysmurf** [comment]
LGTM

**Pull request merged**